### PR TITLE
Refactor Rule assignment task and types

### PR DESCRIPTION
**Pull request by GitEvil**

### CODE REVIEW

File: src/types/ShopwareTypes.ts [hunk 1]
```diff
@@ -242,9 +242,12 @@ export enum RuleType {
     promotionCustomer = 'promotionCustomerRule',
     promotionCart = 'promotionCartRule',
 }
-export interface RuleAssignmentEntity {
+
+export interface RuleAssignmentEntities {
```
Comment (frobel) [inline src/types/ShopwareTypes.ts]: Why do you define it as Entities? Usually it is defining a single entity and as soon as you add an [] Array to it the developer knows okay it is more then one.

Comment (frobel) [reply src/types/ShopwareTypes.ts]: And why you exclude the ruleType from the entity?

**Review by frobel — CHANGES_REQUESTED** [review]

File: src/tasks/shop-admin/Rule/AssignEntitiesToRule.ts [hunk 2]
```diff
@@ -2,11 +2,11 @@ import {Locator, test as base} from '@playwright/test';
 import type {Task} from '../../../types/Task';
 import type {FixtureTypes} from '../../../types/FixtureTypes';
 import {RuleType} from '../../../types/ShopwareTypes';
```
Comment (frobel) [inline src/tasks/shop-admin/Rule/AssignEntitiesToRule.ts]:
```suggestion
import type { Task } from '../../../types/Task';
import type { FixtureTypes } from '../../../types/FixtureTypes';
import { RuleType } from '../../../types/ShopwareTypes';
```

File: src/tasks/shop-admin/Rule/AssignEntitiesToRule.ts [hunk 3]
```diff
@@ -2,11 +2,11 @@ import {Locator, test as base} from '@playwright/test';
...
-import {RuleAssignmentEntity} from '../../../types/ShopwareTypes';
+import {RuleAssignmentEntities} from '../../../types/ShopwareTypes';
```
Comment (frobel) [inline src/tasks/shop-admin/Rule/AssignEntitiesToRule.ts]:
```suggestion
import { RuleAssignmentEntities } from '../../../types/ShopwareTypes';
```

**Review by frobel — CHANGES_REQUESTED** [review]

File: src/types/ShopwareTypes.ts (hunk 1 above)
Comment (GitEvil) [reply src/types/ShopwareTypes.ts]: The ruleType cannot be extracted from the Entity created by the TestDataService but has to be decided by the test, so it is not part of the entity object.

**Review by frobel — APPROVED** [review]

**Pull request merged**